File: src/main/java/com/team3/gudit/sale/service/InventoryService.java
package com.team3.gudit.sale.service;

public interface InventoryService {

    void decreaseStock(Long saleId, Long userId, int quantity);

    void restoreStock(Long saleId, Long userId, int quantity);

    void restoreStockIdempotently(
            String eventId,
            Long saleId,
            Long userId,
            int quantity
    );
}

File: src/main/kotlin/com/team3/gudit/sale/service/InventoryService.kt
package com.team3.gudit.sale.service

interface InventoryService {
    fun decreaseStock(
        saleId: Long?,
        userId: Long?,
        quantity: Int,
    )

    fun restoreStock(
        saleId: Long?,
        userId: Long?,
        quantity: Int,
    )

    fun restoreStockIdempotently(
        eventId: String?,
        saleId: Long?,
        userId: Long?,
        quantity: Int,
    )
}
